src/util-decode-mime.c

```diff
@@ -2660,14 +2666,17 @@ static int TestDataChunkCallback(const uint8_t *chunk, uint32_t len,
         uint8_t *tok;
         uint32_t tokLen;
 
+        if ((*line_count) == 0) {
+            (*line_count)++;
+        }
         PrintChars(SC_LOG_DEBUG, "CHUNK", chunk, len);
 
         /* Parse each line one by one */
         remainPtr = (uint8_t *) chunk;
         do {
             tok = GetLine(remainPtr, len - (remainPtr - (uint8_t *) chunk),
                     &remainPtr, &tokLen);
-            if (tok != NULL && tok != remainPtr) {
+            if (tok != NULL && tok != remainPtr && remainPtr[-1] == LF) {
```

Comment:
Can't say I've ever used a negative array pointer, but it kinda makes sense here.

Reply:
It will make more sense to remove all this code with https://github.com/OISF/suricata/pull/9038 ;-)

Reply:
I didn't know they existed in C 🥲

Reply:
As Mark Twain said : 

> They did not know it was impossible so they did it

Reply:
Reading this code I don't (clearly) see how we can be sure `remainPtr[-1]` accesses valid memory here.

Reply:
> Reading this code I don't (clearly) see how we can be sure remainPtr[-1] accesses valid memory here.

This is just testing code (this is not used by Suricata on live traffic, and not even compiled if you do not take the unit tests), and the tests pass :-p

(By the way, these tests are responsible for having a function pointer, hence the big vulnerability that got fixed around)

I guess I should get rid of the call to `GetLine` and count `\n` more basically...